service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/ReportController.java

```diff
@@ -46,23 +55,58 @@
         produces = TEXT_XML_VALUE)
     @ResponseStatus(value = ACCEPTED)
     public ResponseEntity<String> postReport(@RequestBody String reportXml) {
+        String toAddress = null;
+        String messageId = null;
         try {
             Map<ReportElement, String> reportElementsMap = ReportParserUtil.parseReportXml(reportXml);
-            String messageId = reportElementsMap.get(MESSAGE_ID);
             String trackingId = reportElementsMap.get(TRACKING_ID);
+            messageId = reportElementsMap.get(MESSAGE_ID);
+            toAddress = getValueOrDefaultAddress(reportElementsMap.get(ADDRESS));
+
             LOGGER.info("ITK SOAP message received. MessageId: {}, ItkTrackingId: {}",
                 messageId, trackingId);
+
             POCDMT000002UK01ClinicalDocument1 clinicalDocument = extractClinicalDocument(reportElementsMap
                 .get(DISTRIBUTION_ENVELOPE));
+
             validate(clinicalDocument);
 
             encounterReportService.transformAndPopulateToGP(clinicalDocument,
                 messageId, trackingId);
 
             return new ResponseEntity<>(itkResponseUtil.createSuccessResponseEntity(messageId, randomUUID().toString().toUpperCase()), OK);
-        } catch (XmlException | DocumentException e) {
+        } catch (DocumentException e) {
+            LOGGER.error(BAD_REQUEST.toString() + e.getMessage());
+            throw new ResponseStatusException(BAD_REQUEST, createErrorResponse(
+                toAddress, messageId, CLIENT_ERROR_CODE, "This is not a valid XML message", e.getMessage()));
+        } catch (XmlException e) {
+            LOGGER.error(BAD_REQUEST.toString() + e.getMessage());
+            throw new ResponseStatusException(BAD_REQUEST, createErrorResponse(
+                toAddress, messageId, CLIENT_ERROR_CODE, "Message body not valid", e.getMessage()));
+        } catch (ItkXmlException e) {
```

Comment:
`catch (ItkXmlException | SoapClientException e)` could avoid code duplications. ;)

Reply:
Cannot resolve method 'getReason' in 'Exception'

Reply:
Interesting. What I see both exceptions have this method O.o